UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/GeneratorImpl.cs

```diff
@@ -125,14 +168,8 @@ public class PreviousActionInputGenerator : TensorGenerator.Generator
     {
         public void Generate(Tensor tensor, int batchSize, Dictionary<Agent, AgentInfo> agentInfo)
         {
-            if (tensor.ValueType != Tensor.TensorType.Integer)
-            {
-                throw new NotImplementedException(
-                    "Previous Action Inputs are only valid for discrete control");
```

Comment:
Why was this removed ? It breaks the Unit Tests

Reply:
Barracuda doesn't support Integer tensors, all tensors currently are float32

Reply:
I changed Edit mode GeneratePreviousActionInput test to match new logic